Verification of R package does not use `CMAKE_BIN` environment variable

...As identified in centos7 verification, which sets that to `cmake3`:

https://github.com/apache/arrow-nanoarrow/actions/runs/5166363219/jobs/9306443723#step:4:1837

```
  ** libs
  gcc -m64 -std=gnu99 -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -fpic  -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic  -c altrep.c -o altrep.o
  gcc -m64 -std=gnu99 -I"/usr/include/R" -DNDEBUG   -I/usr/local/include  -fpic  -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic  -c array.c -o array.o
  array.c: In function ‘borrow_buffer’:
  array.c:389:51: error: ‘struct ArrowLayout’ has no member named ‘buffer_data_type’
                                   array_view->layout.buffer_data_type[i],
                                                     ^
```

--- a/r/bootstrap.R
+++ b/r/bootstrap.R
@@ -27,7 +27,7 @@ dist_dir <- file.path(temp_dir, "dist")
 dir.create(build_dir)
 dir.create(dist_dir)
 
-cmake_command <- shQuote(Sys.getenv("CMAKE", "cmake"))
+cmake_command <- shQuote(Sys.getenv("CMAKE_BIN", "cmake"))
 
 run_cmake <- function(args, wd = ".") {
   force(args)

